tests/DGMITLP_test.py

```diff
@@ -112,15 +130,19 @@ def test_dgmitlp(keysize, messages, intervals):
     )
 
     # TPH
-    pk, sk = dgmitlp.helper_setup(intervals, squarigns_per_second_helper)
+    sc.switch_to_account(client_helper_id)
+    pk, sk = dgmitlp.helper_setup(intervals, squarings_per_second_helper)
     puzz_list = dgmitlp.helper_generate(encrypted_messages, pk, sk, start_time, sc)
 
     # TPH'
+    sc.switch_to_account(server_helper_id)
     s = dgmitlp.solve(sc, server_info, pk, puzz_list, coins_acceptable)
     for m_, d in s:
         dgmitlp.register(sc, m_, d)
 
+    # server
+    sc.switch_to_account(server_id)
     for i, message in enumerate(messages):
-        dgmitlp.verify(sc, i)
+        dgmitlp.pay(sc, i)
```

Comment:
this might want a check that `sc.pay` was called for each message

Reply:
Can you please explain what would be the reason to check this?

Reply:
I was just thinking that verify might be failing but now it's not visible

Reply:
Though does not `pay` also do `verify`?

Reply:
so if verify would fail -> pay would fail, or is this too much assumption?

Reply:
```py
    def pay(self, sc: SCInterface, i: int) -> None:
        try:
            self.verify(sc, i)
        except AssertionError:
            sc.pay_back(i)
        else:
            sc.pay(i)
```
So `self.verify` failing causes the coins to be paid back, it won't bubble up

Reply:
Got it, thank you far the clarification, for some reason I thought the error was propagated. Would that work to resolve it (already pushed as a commit)? 

```suggestion
        for i, message in enumerate(messages):
        dgmitlp.verify(sc, i)
        dgmitlp.pay(sc, i)
```